frontend-react/src/components/MessageTracker/MessageReceivers.tsx

```diff
@@ -1,62 +1,511 @@
+    //     "createdAt": "2023-02-03T18:22:55.580322",
+    //     "qualityFilters": []
+    // }
+    // And we parse fileUrl using parseFileLocation to find: folderLocation, sendingOrg and fileName. We use normalizedData to fix this. If there's no data, we fill in the string with N/A.
+
+    // Our expected output should look like the following:
+    //   NormalizedReceiverData {
+    //     Name: string;
+    //     Service: string;
+    //     Date: string;
+    //     ReportId: string;
+    //     Main: string;
+    //     Sub: string;
+    //     FileName: string;
+    //     TransportResults: string;
+    // }
+    const normalizedData = useMemo(
+        () =>
+            receiverDetails.map(
+                (receiverItem: ReceiverData): NormalizedReceiverData => {
+                    const formattedData = Object.keys(ColumnDataEnum).reduce(
+                        (accumulator: any, currentValue: string) => {
+                            accumulator[currentValue] = "N/A";
+                            return accumulator;
+                        },
+                        {}
+                    );
+                    for (const key of Object.keys(ColumnDataEnum)) {
+                        const columnTitle =
+                            ColumnDataEnum[key as keyof typeof ColumnDataEnum];
+                        switch (true) {
+                            case columnTitle === ColumnDataEnum.Name:
+                                if (receiverItem.receivingOrg)
+                                    formattedData.Name =
+                                        receiverItem.receivingOrg;
+                                break;
+                            case columnTitle === ColumnDataEnum.Service:
+                                if (receiverItem.receivingOrgSvc)
+                                    formattedData.Service =
+                                        receiverItem.receivingOrgSvc;
+                                break;
+                            case columnTitle === ColumnDataEnum.Date:
+                                if (receiverItem.createdAt)
+                                    formattedData.Date =
+                                        dateTimeFormatter.format(
+                                            new Date(receiverItem.createdAt)
+                                        );
+
+                                break;
+                            case columnTitle === ColumnDataEnum.ReportId:
+                                if (receiverItem.reportId)
+                                    formattedData.ReportId =
+                                        receiverItem.reportId;
+                                break;
+                            case columnTitle === ColumnDataEnum.Main:
+                                const { folderLocation } = parseFileLocation(
```

Comment:
Similar to [the point above](https://github.com/CDCgov/prime-reportstream/pull/8295/files#r1114986589), we don't need to call `parseFileLocation` three times; we can call it once and destructure from that, no?

Reply:
In code yes, but during runtime, my thinking was that it would actually be called MORE often since if we pulled `parseFileLocation` to the very top, EVERY time `normalizedData` is re-rendered, the `parseFileLocation` would be called instead of just for its proper 3 cases. Is that over thinking it?

Reply:
Ah, nope -- you're right :)  I'm the one overthinking it here 😆